src/onnx-mlir.cpp

```diff
@@ -13,78 +13,29 @@
-      llvm::cl::cat(OnnxMlirOptions));
-
-  llvm::cl::opt<std::string> outputBaseName("o",
-      llvm::cl::desc("Base path for output files, extensions will be added."),
-      llvm::cl::value_desc("path"), llvm::cl::cat(OnnxMlirOptions),
-      llvm::cl::ValueRequired);
-
-  llvm::cl::opt<EmissionTargetType> emissionTarget(
-      llvm::cl::desc("Choose target to emit:"),
-      llvm::cl::values(
-          clEnumVal(EmitONNXBasic,
-              "Ingest ONNX and emit the basic ONNX operations without "
-              "inferred shapes."),
-          clEnumVal(
-              EmitONNXIR, "Ingest ONNX and emit corresponding ONNX dialect."),
-          clEnumVal(EmitMLIR,
-              "Lower the input to MLIR built-in transformation dialect."),
-          clEnumVal(
-              EmitLLVMIR, "Lower the input to LLVM IR (LLVM MLIR dialect)."),
-          clEnumVal(EmitObj, "Compile the input into a object file."),
-          clEnumVal(
-              EmitLib, "Compile the input into a shared library (default)."),
-          clEnumVal(EmitJNI, "Compile the input into a jar file.")),
-      llvm::cl::init(EmitLib), llvm::cl::cat(OnnxMlirOptions));
-
-  // Register MLIR command line options.
-  mlir::registerMLIRContextCLOptions();
-  mlir::registerPassManagerCLOptions();
-  mlir::registerDefaultTimingManagerCLOptions();
-
-  llvm::cl::SetVersionPrinter(getVersionPrinter);
 
-  // Parse options from argc/argv and default ONNX_MLIR_FLAG env var.
-  llvm::cl::ParseCommandLineOptions(argc, argv,
-      "ONNX-MLIR modular optimizer driver\n", nullptr,
-      OnnxMlirEnvOptionName.c_str());
-
-  mlir::OwningOpRef<mlir::ModuleOp> module;
-  std::string errorMessage;
-  int rc = processInputFile(inputFilename, context, module, &errorMessage);
-  if (rc != 0) {
-    if (!errorMessage.empty())
-      std::cerr << errorMessage << std::endl;
-    return 1;
+  const char *errorMessage = NULL;
+  std::string commandLineStr = "";
+  for (int i = 1; i < argc; i++) {
+    commandLineStr.append(std::string(argv[i]).append(" "));
   }
-
-  // Input file base name, replace path if required.
-  // outputBaseName must specify a file, so ignore invalid values
-  // such as ".", "..", "./", "/.", etc.
-  bool b = false;
-  if (outputBaseName == "" ||
-      (b = std::regex_match(
-           outputBaseName.substr(outputBaseName.find_last_of("/\\") + 1),
-           std::regex("[\\.]*$")))) {
-    if (b)
-      std::cerr << "Invalid -o option value " << outputBaseName << " ignored."
-                << std::endl;
-    outputBaseName = inputFilename.substr(0, inputFilename.find_last_of("."));
+  const char *flags = commandLineStr.c_str();
+  int rc = onnx_mlir::omCompileFromFileViaCommand(flags, &errorMessage);
+  if (rc != onnx_mlir::CompilerSuccess) {
+    std::cerr << "Failed to compile options with error code " << rc << "."
+              << std::endl;
+    if (errorMessage) {
+      std::cerr << errorMessage << std::endl;
+    }
+    return rc;
   }
```

Comment:
@Jiaqing-ASU I don't really understand what you are trying to do here. `onnx-mlir.cpp:main()` is the entry point of `onnx-mlir` executable. It's calling `omCompileFromFileViaCommand`, which from what I can see is calling `onnx-mlir` again. So you are asking `onnx-mlir` to call itself again and again.

This is why your PR is failing on all the Jenkins CI servers because they all run out of memory and Jenkins gets killed. On MacOS, it's failing because the system couldn't spawn more process anymore. Windows failure is different. It failed because function `strtok_r` doesn't exist on Windows (we need the Windows equivalent) but it would have failed similarly if `strtok_r` existed.

In any case, the modification to `onnx-mlir.cpp:main()` is completely wrong from what I can see. What you really need is to study a bit on `llvm::cl::opt` related stuff and see if there is any way to put all the options in thread local storage (i.e., each thread gets its own copy of the options).